src/app/clusters/thermostat-server/thermostat-server.cpp

```diff
@@ -0,0 +1,467 @@
+                    status = EMBER_ZCL_STATUS_INVALID_VALUE;
+                break;
+
+            case EMBER_ZCL_THERMOSTAT_CONTROL_SEQUENCE_HEATING_ONLY:
+            case EMBER_ZCL_THERMOSTAT_CONTROL_SEQUENCE_HEATING_WITH_REHEAT:
+                if (RequestedSystemMode == EMBER_ZCL_THERMOSTAT_SYSTEM_MODE_COOL ||
+                    RequestedSystemMode == EMBER_ZCL_THERMOSTAT_SYSTEM_MODE_PRECOOLING)
+                    status = EMBER_ZCL_STATUS_INVALID_VALUE;
+                break;
+            default:
+                break;
+            }
+        }
+    }
+    default:
+        break;
+    }
+
+    return status;
+}
+
+bool emberAfThermostatClusterClearWeeklyScheduleCallback(EndpointId aEndpointId, chip::app::CommandHandler * commandObj)
+{
+    // TODO
+    return false;
+}
+bool emberAfThermostatClusterGetRelayStatusLogCallback(EndpointId aEndpointId, chip::app::CommandHandler * commandObj)
+{
+    // TODO
+    return false;
+}
+
+bool emberAfThermostatClusterGetWeeklyScheduleCallback(EndpointId aEndpointId, chip::app::CommandHandler * commandObj,
+                                                       uint8_t daysToReturn, uint8_t modeToReturn)
+{
+    // TODO
+    return false;
+}
+
+bool emberAfThermostatClusterSetWeeklyScheduleCallback(EndpointId aEndpointId, chip::app::CommandHandler * commandObj,
+                                                       uint8_t numberOfTransitionsForSequence, uint8_t daysOfWeekForSequence,
+                                                       uint8_t modeForSequence, uint8_t * payload)
+{
+    // TODO
+    return false;
+}
+
+using namespace chip::app::Clusters::Thermostat::Attributes;
+
+int16_t EnforceHeatingSetpointLimits(int16_t HeatingSetpoint, EndpointId endpoint)
+{
+    // Optional Mfg supplied limits
+    int16_t AbsMinHeatSetpointLimit = 700;  // 7C (44.5 F) is the default
+    int16_t AbsMaxHeatSetpointLimit = 3000; // 30C (86 F) is the default
+
+    // Optional User supplied limits
+    int16_t MinHeatSetpointLimit = 700;  // 7C (44.5 F) is the default
+    int16_t MaxHeatSetpointLimit = 3000; // 30C (86 F) is the default
+
+    // Attempt to read the setpoint limits
+    // Absmin/max are manufacturer limits
+    // min/max are user imposed min/max
+
+    // Note that the limits are initialized above per the spec limits
+    // if they are not present emberAfReadAttribute() will not update the value so the defaults are used
+    EmberAfStatus status;
+
+    // https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/3724
+    // behavior is not specified when Abs * values are not present and user values are present
+    // implemented behavior accepts the user values without regard to default Abs values.
+
+    // Per global matter data model policy
+    // if a attribute is not present then it's default shall be used.
+
+    status = GetAbsMinHeatSetpointLimit(endpoint, &AbsMinHeatSetpointLimit);
```

Comment:
should we ignore status here and below? In this case we should probably (void) although realistically a log on error (even  if just warning) is probably useful.

Reply:
Added  "Warning: AbsMaxHeatSetpointLimit missing using default"